SoftwareDesign.md

```diff
@@ -0,0 +1,3 @@
+My favorite design pattern I learned this week was the MVC mode. Breaking the model, view, and controller up allows you to focus specifically on the task at hand, and let them do their jobs well. Although Django uses more of a model, view, template(which is also like a controller), and it is a little tricky for myself to get wrap my head around, I still see the advantages and plan on improving my skills int his language and framework. 
+
+An anti pattern that I run into quite often is spaghetti code. I usually solve a problem and then refactor, but I have found myself a few time repeating myself, and writing long functions, and just stuff all over the place. I have started to combat this from the get go but writing concise pseudocode before I even write a line of code. This allows me to take a step by step approach and break the problem up into smaller parts.
```

Comment:
I find that tackling a problem by just throwing code down can be one of the best ways to work quickly, but it's important to be able to combine that skill with good refactor skills to improve the hodgepodge mess into something useful. Keep practicing this!